coalib/bearlib/aspects/Smell.py

```diff
@@ -325,3 +543,46 @@ class docs:
         'This the maximum number of embedded branches or embedded loops'
         ' allowed.',
         (6,), default=6)
+
+
+@Complexity.subaspect
+class MaintainabilityIndex:
+    """
+    The `maintainablity index
+    <www.projectcodemeter.com/cost_estimation/help/GL_maintainability.htm>`_
+    is a software metric which measures how
+    maintanable a source code is.
+
+    The maintainability index is always in the range 0-100 and is ranked
+    as follow:\n
+    * A `MI` in the range 0-9 maps to a code extremely difficult to maintain.
+    * A `MI` in the range 10-19 maps to a maintainable code.
+    * A `MI` in the range 20-100 maps to a code highly maintainable.
+    """
+    class docs:
+        example = """
+        '''
+        The maintainability index of our `linter` class along with
+        the `_create_linter` and `_prepare_options` functions found `Here
+        <https://github.com/coala/coala/blob/master/coalib/bearlib/abstractions
+        /Linter.py>`_ is 51.76 which is very good.
+        The maintainability index for the following piece of code is 100.
+        '''
+        from coalib.bearlib.aspects import Root
+
+        def traverse_aspect_tree(aspect):
+            for aspects in aspect.subaspects.values():
+                print(aspect.__name__, aspect.docs.definition)
+                traverse_aspect_tree(aspects)
+        traverse_aspect_tree(Root)
+        """
+        example_language = 'python'
+        importance_reason = """
+        Complex codes are difficult to maintain.
+        """
+        fix_suggestions = """
+        This can be solved by writing simpler functions and methods.
+        """
+    maintainability_index = Taste[int](
+        'Maintainability index of your code',
+        (9, 19), default=19)
```

Comment:
changed it to 10 as 10 means that your code is maintainable below :-1: above :+1: